frontend/src/components/MinioArtifactPreview.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { stylesheet } from 'typestyle';
+import { color } from '../Css';
+import { StorageService } from '../lib/WorkflowParser';
+import { S3Artifact } from '../../third_party/argo-ui/argo_template';
+import { isS3Endpoint } from '../lib/AwsHelper';
+import { Apis } from '../lib/Apis';
+import { ExternalLink } from '../atoms/ExternalLink';
+import { ValueComponentProps } from './DetailsTable';
+
+const css = stylesheet({
+  root: {
+    width: '100%',
+    whiteSpace: 'nowrap',
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+  },
+  preview: {
+    maxHeight: 250,
+    overflowY: 'auto',
+    padding: 3,
+    backgroundColor: color.lightGrey,
+  },
+});
+
+/**
+ * Check if a javascript object is an argo S3Artifact object.
+ * @param value Any javascript object.
+ */
+export function isS3Artifact(value: any): value is S3Artifact {
+  return value && value.key && value.bucket;
+}
+
+export interface MinioArtifactPreviewProps extends ValueComponentProps<Partial<S3Artifact>> {
+  namespace?: string;
+  peek?: number;
+}
+
+function getStoragePath(value?: string | Partial<S3Artifact>) {
+  if (!value || typeof value === 'string') return;
+  const { key, bucket, endpoint } = value;
+  if (!bucket || !key) return;
+  const source = isS3Endpoint(endpoint) ? StorageService.S3 : StorageService.MINIO;
+  return { source, bucket, key };
+}
+
+/**
+ * A component that renders a preview to an artifact with a link to the full content.
+ */
+const MinioArtifactPreview: React.FC<MinioArtifactPreviewProps> = ({
+  value,
+  namespace,
+  peek = 255,
+}) => {
+  const [content, setContent] = React.useState<string | undefined>(undefined);
+  const storagePath = getStoragePath(value);
+
+  React.useEffect(() => {
+    let cancelled = false;
+    if (storagePath) {
+      // TODO how to handle binary data
+      Apis.readFile(storagePath, namespace, peek + 1).then(
+        data => !cancelled && setContent(data.length > peek ? `${data.slice(0, peek)} ...` : data),
```

Comment:
Later we can have more advanced logic here. For example <=10 full lines or <= 10KB whichever comes first.

P.S. The ellipsis is a nice touch (maybe later we can put it on its own line)

Reply:
+1 to putting `...` on its own line, why not do it now?